src/backend/joanie/core/models/products.py

```diff
@@ -494,3 +473,20 @@ def set(self):
                 self.state_fail(f'Enrollment failed for course run "{link:s}".')
             else:
                 self.state = enums.ENROLLMENT_STATE_SET
+
+
+@receiver(models.signals.post_delete, sender=Enrollment)
+# pylint: disable=unused-argument
+def enrollment_post_delete_receiver(sender, instance, *args, **kwargs):
+    """
+    Post delete receiver method for enrollment method.
+
+    Each time we delete an enrollment from database, we have also to unenroll
```

Comment:
ok but what is the point of allowing to delete enrollments? Maybe it's better to forbid this and mark it inactive the same way as in OpenEdX?

Reply:
In order to try to keep database clean, I currently remove enrollments on order deletion.

https://github.com/openfun/joanie/blob/2a13f0f419de72092380247ab5792f46ff7c0216/src/backend/joanie/core/models/products.py#L281-L290